app/models/category.rb

```diff
@@ -1118,6 +1118,46 @@ def has_restricted_tags?
     tags.count > 0 || tag_groups.count > 0
   end
 
+  def self.reorder_categories!
+    categories =
+      if SiteSetting.fixed_category_positions
+        Category.order(:position)
+      else
+        Category
+          .left_outer_joins(:featured_topics)
+          .group("categories.id")
+          .order("max(topics.bumped_at) DESC NULLS LAST")
```

Comment:
The worry I have here is that this query will still be slow on a site with many categories as long as the ordering condition continues to be complex. For the use cases we are optimising for, will ordering by an existing column on the categories table help?